### ISSUE TITLE
Cannot read property 'rotation' of undefined

### ISSUE THREAD

**Issue by croccio** [mozilla/pdf.js#12829]
Attach (recommended) or Link to PDF file here: all pdf file

Configuration:
- Web browser and its version: Chrome 87.0.4280.88, Opera, Safari and other
- Operating system and its version: Mac os 11.0.1 and windows
- PDF.js version:2.6.347
- Is a browser extension: NO

Steps to reproduce the problem:
1. Zoom continuosly in and out

What is the expected behavior?  Zoom in and out hsould work correctly

What went wrong? it crash

```×
TypeError: Cannot read property 'rotation' of undefined
PDFPageView.cssTransform
node_modules/pdfjs-dist/web/pdf_viewer.js:5207
  5204 | var div = this.div;
  5205 | target.style.width = target.parentNode.style.width = div.style.width = Math.floor(width) + "px";
  5206 | target.style.height = target.parentNode.style.height = div.style.height = Math.floor(height) + "px";
> 5207 | var relativeRotation = this.viewport.rotation - this.paintedViewportMap.get(target).rotation;
       | ^  5208 | var absRotation = Math.abs(relativeRotation);
  5209 | var scaleX = 1,
  5210 |     scaleY = 1;
View compiled
PDFPageView.update
node_modules/pdfjs-dist/web/pdf_viewer.js:5171
  5168 | }
  5169 | 
  5170 | if (this.zoomLayer) {
> 5171 |   this.cssTransform(this.zoomLayer.firstChild);
       | ^  5172 | }
  5173 | 
  5174 | this.reset(true, true);
View compiled
PDFViewer._setScaleUpdatePages
node_modules/pdfjs-dist/web/pdf_viewer.js:6341
  6338 | }
  6339 | 
  6340 | for (var i = 0, ii = this._pages.length; i < ii; i++) {
> 6341 |   this._pages[i].update(newScale);
       | ^  6342 | }
  6343 | 
  6344 | this._currentScale = newScale;
View compiled
PDFViewer._setScale
node_modules/pdfjs-dist/web/pdf_viewer.js:6381
  6378 | var scale = parseFloat(value);
  6379 | 
  6380 | if (scale > 0) {
> 6381 |   this._setScaleUpdatePages(scale, value, noScroll, false);
       | ^  6382 | } else {
  6383 |   var currentPage = this._pages[this._currentPageNumber - 1];
  6384 | 
View compiled
PDFViewer.set
node_modules/pdfjs-dist/web/pdf_viewer.js:6947
  6944 |       return;
  6945 |     }
  6946 | 
> 6947 |     this._setScale(val, false);
       | ^  6948 |   }
  6949 | }, {
  6950 |   key: "currentScaleValue",
View compiled
PdfViewer._this.changeZoom [as onZoomChanged]
src/pages/ContractDetails/component/pdfViewer/PdfViewer.js:48
  45 |     this.enableZoom(false)
  46 |     const pdfViewer = document.getElementsByClassName("PdfHighlighter")[0]
  47 |     if (pdfViewer) {
> 48 |         window.PdfViewer.viewer.currentScale = zoom
     | ^  49 |     }
  50 | }
  51 | 
View compiled
PdfController._this.zoomBy
src/pages/ContractDetails/component/pdfViewer/pdfController/PdfController.js:105
  102 | zoomBy = (zoom) => {
  103 |     if (zoom <= this.maxZoom().value && zoom >= this.minZoom().value) {
  104 |         this.setZoom(zoom);
> 105 |         this.props.onZoomChanged(zoom / 100);
      | ^  106 |     } else if (zoom.toString() !== "")
  107 |         this.setZoom(this.state.zoom);
  108 | }
View compiled
PdfController._this.goToPrevZoom
src/pages/ContractDetails/component/pdfViewer/pdfController/PdfController.js:99
   96 | }
   97 | 
   98 | goToPrevZoom = () => {
>  99 |     this.zoomBy(this.state.zoom - 10);
      | ^  100 | }
  101 | 
  102 | zoomBy = (zoom) => {
View compiled
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js:189
  186 |     window.event = windowEvent;
  187 |   }
  188 | 
> 189 |   func.apply(context, funcArgs);
      | ^  190 |   didError = false;
  191 | } // Create a global error event handler. We use this to capture the value
  192 | // that was thrown. It's possible that this error handler will fire more
View compiled
invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js:238
  235 | // errors, it will trigger our global error handler.
  236 | 
  237 | evt.initEvent(evtType, false, false);
> 238 | fakeNode.dispatchEvent(evt);
      | ^  239 | 
  240 | if (windowEventDescriptor) {
  241 |   Object.defineProperty(window, 'event', windowEventDescriptor);
View compiled
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js:291
  288 | function invokeGuardedCallback(name, func, context, a, b, c, d, e, f) {
  289 |   hasError = false;
  290 |   caughtError = null;
> 291 |   invokeGuardedCallbackImpl$1.apply(reporter, arguments);
      | ^  292 | }
  293 | /**
  294 |  * Same as invokeGuardedCallback, but instead of returning an error, it stores
View compiled
invokeGuardedCallbackAndCatchFirstError
node_modules/react-dom/cjs/react-dom.development.js:306
  303 | 
  304 | 
  305 | function invokeGuardedCallbackAndCatchFirstError(name, func, context, a, b, c, d, e, f) {
> 306 |   invokeGuardedCallback.apply(this, arguments);
      | ^  307 | 
  308 |   if (hasError) {
  309 |     var error = clearCaughtError();
View compiled
executeDispatch
node_modules/react-dom/cjs/react-dom.development.js:391
  388 | function executeDispatch(event, listener, inst) {
  389 |   var type = event.type || 'unknown-event';
  390 |   event.currentTarget = getNodeFromInstance(inst);
> 391 |   invokeGuardedCallbackAndCatchFirstError(type, listener, undefined, event);
      | ^  392 |   event.currentTarget = null;
  393 | }
  394 | /**
View compiled
executeDispatchesInOrder
node_modules/react-dom/cjs/react-dom.development.js:416
  413 |     executeDispatch(event, dispatchListeners[i], dispatchInstances[i]);
  414 |   }
  415 | } else if (dispatchListeners) {
> 416 |   executeDispatch(event, dispatchListeners, dispatchInstances);
      | ^  417 | }
  418 | 
  419 | event._dispatchListeners = null;
View compiled
executeDispatchesAndRelease
node_modules/react-dom/cjs/react-dom.development.js:3301
  3298 | 
  3299 | var executeDispatchesAndRelease = function executeDispatchesAndRelease(event) {
  3300 |   if (event) {
> 3301 |     executeDispatchesInOrder(event);
       | ^  3302 | 
  3303 |     if (!event.isPersistent()) {
  3304 |       event.constructor.release(event);
View compiled
executeDispatchesAndReleaseTopLevel
node_modules/react-dom/cjs/react-dom.development.js:3310
  3307 | };
  3308 | 
  3309 | var executeDispatchesAndReleaseTopLevel = function executeDispatchesAndReleaseTopLevel(e) {
> 3310 |   return executeDispatchesAndRelease(e);
       | ^  3311 | };
  3312 | 
  3313 | function runEventsInBatch(events) {
View compiled
forEachAccumulated
node_modules/react-dom/cjs/react-dom.development.js:3282
  3279 |   if (Array.isArray(arr)) {
  3280 |     arr.forEach(cb, scope);
  3281 |   } else if (arr) {
> 3282 |     cb.call(scope, arr);
       | ^  3283 |   }
  3284 | }
  3285 | /**
View compiled
runEventsInBatch
node_modules/react-dom/cjs/react-dom.development.js:3327
  3324 |   return;
  3325 | }
  3326 | 
> 3327 | forEachAccumulated(processingEventQueue, executeDispatchesAndReleaseTopLevel);
       | ^  3328 | 
  3329 | if (!!eventQueue) {
  3330 |   {
View compiled
runExtractedPluginEventsInBatch
node_modules/react-dom/cjs/react-dom.development.js:3537
  3534 | 
  3535 | function runExtractedPluginEventsInBatch(topLevelType, targetInst, nativeEvent, nativeEventTarget, eventSystemFlags) {
  3536 |   var events = extractPluginEvents(topLevelType, targetInst, nativeEvent, nativeEventTarget, eventSystemFlags);
> 3537 |   runEventsInBatch(events);
       | ^  3538 | }
  3539 | 
  3540 | function handleTopLevel(bookKeeping) {
View compiled
handleTopLevel
node_modules/react-dom/cjs/react-dom.development.js:3581
  3578 |       eventSystemFlags |= IS_FIRST_ANCESTOR;
  3579 |     }
  3580 | 
> 3581 |     runExtractedPluginEventsInBatch(topLevelType, targetInst, nativeEvent, eventTarget, eventSystemFlags);
       | ^  3582 |   }
  3583 | }
  3584 | 
View compiled
batchedEventUpdates$1
node_modules/react-dom/cjs/react-dom.development.js:21729
  21726 | executionContext |= EventContext;
  21727 | 
  21728 | try {
> 21729 |   return fn(a);
        | ^  21730 | } finally {
  21731 |   executionContext = prevExecutionContext;
  21732 | 
View compiled
batchedEventUpdates
node_modules/react-dom/cjs/react-dom.development.js:798
  795 | isBatchingEventUpdates = true;
  796 | 
  797 | try {
> 798 |   return batchedEventUpdatesImpl(fn, a, b);
      | ^  799 | } finally {
  800 |   isBatchingEventUpdates = false;
  801 |   finishEventHandler();
View compiled
dispatchEventForLegacyPluginEventSystem
node_modules/react-dom/cjs/react-dom.development.js:3591
  3588 | try {
  3589 |   // Event queue being processed in the same cycle allows
  3590 |   // `preventDefault`.
> 3591 |   batchedEventUpdates(handleTopLevel, bookKeeping);
       | ^  3592 | } finally {
  3593 |   releaseTopLevelCallbackBookKeeping(bookKeeping);
  3594 | }
View compiled
attemptToDispatchEvent
node_modules/react-dom/cjs/react-dom.development.js:4311
  4308 | }
  4309 | 
  4310 | {
> 4311 |   dispatchEventForLegacyPluginEventSystem(topLevelType, eventSystemFlags, nativeEvent, targetInst);
       | ^  4312 | } // We're not blocked on anything.
  4313 | 
  4314 | return null;
View compiled
dispatchEvent
node_modules/react-dom/cjs/react-dom.development.js:4232
  4229 |   return;
  4230 | }
  4231 | 
> 4232 | var blockedOn = attemptToDispatchEvent(topLevelType, eventSystemFlags, container, nativeEvent);
       | ^  4233 | 
  4234 | if (blockedOn === null) {
  4235 |   // We successfully dispatched this event.
View compiled
unstable_runWithPriority
node_modules/scheduler/cjs/scheduler.development.js:659
  656 | currentPriorityLevel = priorityLevel;
  657 | 
  658 | try {
> 659 |   return eventHandler();
      | ^  660 | } finally {
  661 |   currentPriorityLevel = previousPriorityLevel;
  662 | }
View compiled
runWithPriority$1
node_modules/react-dom/cjs/react-dom.development.js:11077
  11074 | 
  11075 | function runWithPriority$1(reactPriorityLevel, fn) {
  11076 |   var priorityLevel = reactPriorityToSchedulerPriority(reactPriorityLevel);
> 11077 |   return Scheduler_runWithPriority(priorityLevel, fn);
        | ^  11078 | }
  11079 | 
  11080 | function scheduleCallback(reactPriorityLevel, callback, options) {
View compiled
discreteUpdates$1
node_modules/react-dom/cjs/react-dom.development.js:21746
  21743 | 
  21744 | try {
  21745 |   // Should this
> 21746 |   return runWithPriority$1(UserBlockingPriority$1, fn.bind(null, a, b, c, d));
        | ^  21747 | } finally {
  21748 |   executionContext = prevExecutionContext;
  21749 | 
View compiled
discreteUpdates
node_modules/react-dom/cjs/react-dom.development.js:811
  808 | isInsideEventHandler = true;
  809 | 
  810 | try {
> 811 |   return discreteUpdatesImpl(fn, a, b, c, d);
      | ^  812 | } finally {
  813 |   isInsideEventHandler = prevIsInsideEventHandler;
  814 | 
View compiled
dispatchDiscreteEvent
node_modules/react-dom/cjs/react-dom.development.js:4211
  4208 | 
  4209 | function dispatchDiscreteEvent(topLevelType, eventSystemFlags, container, nativeEvent) {
  4210 |   flushDiscreteUpdatesIfNeeded(nativeEvent.timeStamp);
> 4211 |   discreteUpdates(dispatchEvent, topLevelType, eventSystemFlags, container, nativeEvent);
       | ^  4212 | }
  4213 | 
  4214 | function dispatchUserBlockingUpdate(topLevelType, eventSystemFlags, container, nativeEvent) {
View compiled
This screen is visible only in development. It will not appear if the app crashes in production.
Open your browser’s developer console to further inspect this error.```

**Comment by Snuffleupagus** [comment]
Please see https://github.com/mozilla/pdf.js/blob/master/.github/CONTRIBUTING.md (emphasis mine):
> If you are developing a custom solution, first check the examples at https://github.com/mozilla/pdf.js#learning and search existing issues. If this does not help, **please prepare a short well-documented example that demonstrates the problem and make it accessible online** on your website, JS Bin, GitHub, etc. before opening a new issue or contacting us in the Matrix room -- **keep in mind that just code snippets won't help us troubleshoot the problem**.
